client/pages/your-listing-details.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import Redirect from '../components/redirect';
+
+export default class YourListingDetails extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      listing: null,
+      token: this.props.token
+    };
+  }
+
+  componentDidMount() {
```

Comment:
You should include a guard in here that prevents the fetch if there's no user or token.